src/routers/threadRouter.js

```diff
@@ -11,6 +11,7 @@ threadRouter.get('/:id', threadController.getThread);
 
 // POST APIS
 threadRouter.post('/', processPet, threadController.createThread);
+threadRouter.post('/:id/favorite', threadController.favoriteThread);
```

Comment:
why is it a post? if u are adding a thread id to a user's favorites field, then it is a patch of a put.

Reply:
I am not using a User router but a thread router, so I only push or remove the favorite threadId to or from the User favored thread list after the thread router checks whether this thread is existed and has been favorited before or not. 

Under thread router,  if a thread has been favorite or unfavorite, I choose to post, as the favorite is not a required schema in our design. So I think I cannot update or patch a prop that probably not existed before. I guess if we all agree favorite is a required schema, then I can use a patch of a put.

Reply:
Let me know if I am not clear or understand something wrongly. Thx

Reply:
Like ultimately you are still altering user right, you are adding/removing a user's favorites, which does not create any new resource in the backend, you are just altering an already existing resource.